x-pack/plugins/console_extensions/server/routes/api/text_object/with_current_username.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { RequestHandler } from 'kibana/server';
+import { Authentication } from '../../../../../security/server';
+
+interface WithCurrentUsernameArgs {
+  /**
+   * Security provided {@link Authentication} for securely determining user information.
+   */
+  authc: Authentication;
+
+  /**
+   * Handler factory: a function that we can call to return a {@link RequestHandler}.
+   *
+   * This wrapper will inject authenticated user information to the handler.
+   *
+   * @remark
+   * Ideally we want to have a type like <P, Q, B>(username: string) => RequestHandler<P, Q, B>.
+   *
+   * However, when assigning with body like in `create.ts` TS complains that B
+   * not assignable to `ReadOnly<{...}>`.
+   */
+  handlerFactory: (username: string) => any;
+}
+
+export const withCurrentUsername = ({
+  authc,
+  handlerFactory,
+}: WithCurrentUsernameArgs): RequestHandler => async (ctx, request, response) => {
+  const authenticatedUser = await authc.getCurrentUser(request);
```

Comment:
When security is disabled in Elasticsearch, this will throw an `Error`. So users just starting off with the Basic license who haven't gotten around to enabling security yet will be unable to use dev tools and see the following error:

<img width="1059" alt="Screen Shot 2020-02-07 at 4 08 56 PM" src="https://user-images.githubusercontent.com/627123/74074707-2f0cb100-49c4-11ea-8bb9-aae25aafd43c.png">

Reply:
Thanks for testing this case! I was still getting around to testing these permutations. We will need to perform a check before registering a Saved Objects adapter.